addons/danger/functions/fnc_leaderStaticPack.sqf

```diff
@@ -0,0 +1,122 @@
+#include "script_component.hpp"
+/*
+ * Author: nkenny
+ * Pack static weapon
+ *
+ * Arguments:
+ * 0: Units <ARRAY>, <OBJECT>, <GROUP>
+ * 1: Range check <NUMBER>
+ * 2: Weapon to pack <ARRAY>
+ *
+ * Return Value:
+ * units in array
+ *
+ * Example:
+ * [units bob] call lambs_danger_fnc_leaderStaticPack;
+ *
+ * Public: No
+*/
+
+params ["_units", ["_guns", []]];
+
+// sort units
+if (_units isEqualType objNull) then { _units = [_units] call FUNC(findReadyUnits); };
+if (_units isEqualType grpNull) then { _units = [leader _units] call FUNC(findReadyUnits); };
+
+// get weapons list
+if (_guns isEqualTo []) then {
+    _guns = (group (_units select 0)) getVariable [QGVAR(staticWeaponList), []];
+};
+
+// check if weapon is unmanned
+_guns = _guns select {!(crew _x isEqualTo [])};
+if (_guns isEqualTo []) exitWith { _units };
+
+// get gunner
+private _weapon = (_guns deleteAt 0);
+private _gunner = gunner _weapon;
+
+// check for nearest unit without backpack
+_units = _units select { isNull objectParent _x && { (backpack _x) isEqualTo ""} };
+
+// get nearest assistant ~ should probably write a function for this! - nkenny
+if (_units isEqualTo []) exitWith { _units };
+_units = _units apply { [ _x distance2d _gunner, _x ] };
+_units sort true;
+
+// get assistant
+private _assistant = (_units select 0) select 1;
+
+// eventhandler ~ inspired by BIS_fnc_unpackStaticWeapon by Rocket and Killzone_Kid
+_gunner setVariable [QGVAR(staticWeaponAssistant), _assistant];
+private _EH = _gunner addEventHandler ["WeaponDisassembled", {
+        params ["_gunner", "_weaponBag", "_baseBag"];
+
+        // get assistant
+        private _assistant = _gunner getVariable [QGVAR(staticWeaponAssistant), selectRandom units _gunner];
```

Comment:
assistent can get null if the assistent dies. in this case we should select a unit that does not have a backpack
i would return objNull here as default and check for if isNull and than select a unit that does not have a backpack

Reply:
I figured selecting a random one is adequate.  If the unit 'fails' to collect a backpack, because assistant is dead, or looses a backpack from another unit in the group. Then that is fine.  It is all part of the verisimilitude that is A3 fire fights. 

Not even humans are perfect at collecting these things.

Reply:
if the assistant the variable will return objNull and not nil so the default will never trigger.

Reply:
Actually. If assistant fails in the process. the CBA waitUntilAndExecute, will exit the script in any case.

Reply:
then is still no default required.